### PR TITLE
Invited members in room's members list

### PR DESCRIPTION
**Pull request by alfogrillo**
This PR:
- Optimizes the performances of the room's details by moving heavy operations in a background thread
- Shows invited people in the room's members list

Fix https://github.com/vector-im/element-x-ios/issues/725

**Result**
demo

### CODE REVIEW

File: ElementX/Sources/Screens/RoomDetailsScreen/RoomDetailsScreenViewModel.swift [hunk 1]
```diff
@@ -96,6 +78,49 @@ class RoomDetailsScreenViewModel: RoomDetailsScreenViewModelType, RoomDetailsScr
...
+                
+                if member.membership == .join {
+                    joinedMembersCount += 1
+                }
+            }
+            
+            return (roomMembersDetails, joinedMembersCount)
+        }
+        .value
+    }
```
Comment (Velin92) [inline ElementX/Sources/Screens/RoomDetailsScreen/RoomDetailsScreenViewModel.swift:122]: I am wondering, if we remove the `@MainActor` property from the RoomMmemberProxyProtocol, and we just have the functions that are required to be MainActor, does this fix the slowdown issue? I think we can make only the RoomMemberProxy constructor to be MainActor.

Comment (alfogrillo) [reply ElementX/Sources/Screens/RoomDetailsScreen/RoomDetailsScreenViewModel.swift:122]: Already did [here](https://github.com/vector-im/element-x-ios/pull/880/files#diff-e5ac784f064b5fb41928ce11783ba46e778574ff0b5174ba8b302ecf405fc548L25) and also in the `RoomMemberDetails`'s init.
It looks helping a lot for performances. ;)

File: UITests/Sources/__Snapshots__/Application/pseudo-iPhone-14.roomMembersListScreenPendingInvites-1.png
Comment (Velin92) [inline UITests/Sources/__Snapshots__/Application/pseudo-iPhone-14.roomMembersListScreenPendingInvites-1.png:1]: Uhm I think we should dismiss the keyboard in these text I am not sure the CI is going to have the same keyboard pattern/suggestions making the test fail on the CI

**Review by Velin92 — APPROVED** [review]
LGTM just be sure to dismiss the keyboard in the UI tests, so that they are CI safe

File: UITests/Sources/__Snapshots__/Application/pseudo-iPhone-14.roomMembersListScreenPendingInvites-1.png
Comment (alfogrillo) [reply UITests/Sources/__Snapshots__/Application/pseudo-iPhone-14.roomMembersListScreenPendingInvites-1.png:1]: Done: https://github.com/vector-im/element-x-ios/pull/880/commits/809272522426a8effa39f5d0262f037d996d03df

**Pull request merged**